spec/com/adgoji/common.clj

```diff
@@ -5,7 +5,18 @@
 ;;; Constants
 
 (def ^:private email-regex #"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,63}$")
-(def interval-regex #"^\d+\s(months|weeks|days)")
+(def interval-regex #"^\d+\s(months*|weeks*|days*)")
+
+;;; Helpers
+
+(defmacro only-keys
+  [& {:keys [req req-un opt opt-un] :as args}]
+  `(s/merge (s/map-of ~(set (concat req
+                                    (map (comp keyword name) req-un)
+                                    opt
+                                    (map (comp keyword name) opt-un)))
+                      any?)
+            (s/keys ~@(apply concat (vec args)))))
```

Comment:
I'm probably missing something, but this does something different than `(s/keys :req ...)`. It is allowing strings as well?

Reply:
Here's the macroexpand result for `::subscriptions-list` spec.

<img width="374" alt="image" src="https://github.com/AdGoji/mollie/assets/100711682/2b7d544a-9dd1-4ae7-b28c-899bc65aab10">

The idea is taken from [this](https://groups.google.com/g/clojure/c/fti0eJdPQJ8) thread.